R/estimate_KM.R

```diff
@@ -65,101 +65,68 @@
-    df <- the_lhs()
-    Call[["data"]] <- as.symbol(df)
-  } else {
-     df <- as.character(sub("\\[.*$", "", deparse(dfExpr))[1])
-  }
+  dots <- rlang::dots_list(...)
 
-  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) stop(paste0("Data can be of `class` dataframe or tibble or data.table."))
+  # Validate argument inputs ---------------------------------------------------
 
-  data <- as.data.frame(data)
+  if (is.null(data))
+    stop(paste0("Data can't be NULL."))
 
-# Validate columns --------------------------------------------------------
+  if (!is.data.frame(data))
+    stop("Data does not have class `data.frame`.")
 
   reqcols <- c(strata, CNSR, AVAL)
-
   if (! all(reqcols %in% colnames(data))){
     stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols, colnames(data)), collapse = " "), "."))
   }
 
-  if (! is.numeric(data[[AVAL]])){
+  if (!is.numeric(data[[AVAL]])){
     stop("Analysis variable (AVAL) is not numeric.")
   }
 
-  if (! is.numeric(data[[CNSR]])){
+  if (!is.numeric(data[[CNSR]])){
     stop("Censor variable (CNSR) is not numeric.")
   }
 
-# Ensure the presence of at least one strata -----------------------------
-
-  if (is.null(strata)){
-    main <- "1"
-  } else {
-    main <- paste(strata, collapse = " + ")
-  }
-
-# Remove NA from the analysis --------------------------------------------
+  # Remove NA from the analysis ------------------------------------------------
 
-  data <- as.data.frame(data) %>%
-    tidyr::drop_na(AVAL, CNSR)
+  data <-
+    as.data.frame(data) %>%
+    tidyr::drop_na(any_of(c(AVAL, CNSR)))
 
   if (!is.null(strata)){
     data <- data %>%
-      tidyr::drop_na(any_of({{strata}}))
+      tidyr::drop_na(any_of(strata))
   }
 
-# Calculate survival and add time = 0 to survfit object -------------------
+  # Ensure the presence of at least one strata -----------------------------
 
- ## Reverse censoring: see ADaM guidelines versus R survival KM analysis
+  formula_rhs <-
+    ifelse(is.null(strata), "1", paste(strata, collapse = " + "))
 
-  formula <- stats::as.formula(paste0("survival::Surv(", AVAL, ", 1-", CNSR, ") ~ ", main))
+  # Calculate survival and add time = 0 to survfit object -------------------
 
-  survfit_object <- survival::survfit(
-    formula, data = data, ...
-  )
+  formula <- stats::as.formula(paste0("survival::Surv(", AVAL, ", 1-", CNSR, ") ~ ", formula_rhs))
 
-  survfit_object <- survival::survfit0(
-    survfit_object, start.time = 0
-  )
+  survfit_object <-
+    rlang::inject(survival::survfit(!!formula, data = data, !!!dots)) %>% # immediate resolves call arguments
+    survival::survfit0(start.time = 0)
 
+  # convert survfit() call to quo with attached envir --------------------------
 
-# Update Call with original info and dots, similar as update.default ------
+  survfit_object$call[[1]] <- rlang::expr(survival::survfit) # adding `survival::` prefix
```

Comment:
survfit_object$call[[1]] <- rlang::expr(survival::survfit) # adding `survival::` prefix
  survfit_object$call <- rlang::quo(!!survfit_object$call)

When looking into detail I see the first line is overwritten by the second line of code. One aesthetic issue that bothers me is that the quosure is creating ~ in front of the formula. Do you know why that is introduced? It will confuse less experienced programmers.

Reply:
The first line just adds the survival prefix to the call ( which we can remove, I left it in as it previously appeared ), the second converts the call to a quosure. Is there an issue with the results?

Reply:
Looking at it again, it makes more sense.